Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs

```diff
@@ -292,10 +293,87 @@ await Http.DownloadAsync(x.PluginNewUserPlugin.UrlDownload, downloadToFilePath)
                         ContextData =
                             new UserPlugin
                             {
-                                Website = x.PluginNewUserPlugin.Website, UrlSourceCode = x.PluginNewUserPlugin.UrlSourceCode
+                                Website = x.PluginNewUserPlugin.Website,
+                                UrlSourceCode = x.PluginNewUserPlugin.UrlSourceCode
                             }
                     });
 
+            if (resultsForUpdate.Count() > 1)
+            {
+                var updateAllResult = new Result
+                {
+                    Title = Context.API.GetTranslation("plugin_pluginsmanager_update_all_title"),
+                    SubTitle = Context.API.GetTranslation("plugin_pluginsmanager_update_all_subtitle"),
+                    IcoPath = icoPath,
+                    Action = e =>
+                    {
+                        string message;
+                        if (Settings.AutoRestartAfterChanging)
+                        {
+                            message = string.Format(Context.API.GetTranslation("plugin_pluginsmanager_update_all_prompt"),
+                                                    resultsForUpdate.Count(), Environment.NewLine);
+                        }
+                        else
+                        {
+                            message = string.Format(Context.API.GetTranslation("plugin_pluginsmanager_update_all_prompt_no_restart"),
+                                                    resultsForUpdate.Count());
+                        }
+
+                        if (MessageBox.Show(message,
+                            Context.API.GetTranslation("plugin_pluginsmanager_update_title"),
+                            MessageBoxButton.YesNo) == MessageBoxResult.Yes)
+                        {
+                            Parallel.ForEach(resultsForUpdate, plugin =>
```

Comment:
Use the async version instead as this is mainly io

Reply:
So I should rather use await Task.WhenAll?
because then I get this error and I don't quite know how to solve it.
```
Cannot convert asynchronous lambda expression to delegate type 'Func<ActionContext, bool>'. An asynchronous lambda expression can return void, Task or Task<T>, none of which are convertible to 'Func<ActionContext, bool>'.
```